src/components/svgComponent.ts

```diff
@@ -0,0 +1,354 @@
+/**
+ * Copyright 2014-present Palantir Technologies
+ * @license MIT
+ */
+
+import * as d3 from "d3";
+
+import { AbstractComponent, IComponent } from "./abstractComponent";
+import { Point, SpaceRequest, Bounds } from "../core/interfaces";
+import * as RenderController from "../core/renderController";
+import * as Utils from "../utils";
+
+// HACKHACK replace with GenericComponentCallback in 3.0
+export type ComponentCallback = (component: SVGComponent) => void;
+
+export class SVGComponent extends AbstractComponent<d3.Selection<void>> {
+  protected _boundingBox: d3.Selection<void>;
+  private _backgroundContainer: d3.Selection<void>;
+  private _foregroundContainer: d3.Selection<void>;
+  protected _clipPathEnabled = false;
+
+  private _boxes: d3.Selection<void>[] = [];
+  private _boxContainer: d3.Selection<void>;
+  private _rootSVG: d3.Selection<void>;
+  private _clipPathID: string;
+  private static _SAFARI_EVENT_BACKING_CLASS = "safari-event-backing";
+
+  public constructor() {
+    super();
+    this._cssClasses.add("component");
+  }
+
+  /**
+   * Attaches the Component as a child of a given d3 Selection.
+   *
+   * @param {d3.Selection} selection.
+   * @returns {Component} The calling Component.
+   */
+  public anchor(selection: d3.Selection<void>) {
+    if (this._destroyed) {
+      throw new Error("Can't reuse destroy()-ed Components!");
+    }
+
+    /**
+     * The user has the option of passing an SVG element directly
+     * to the anchor method, in which case we will render directly
+     * to that SVG.
+     */
+    const isTopLevelSVG = (<Element> selection.node()).nodeName.toLowerCase() === "svg";
+    if (isTopLevelSVG) {
+      this._rootSVG = selection;
+    } else {
+      this._rootSVG = selection.append("svg");
```

Comment:
maybe i don't understand how "anchoring" works but does this mean that if you try to "anchor" an `SVGComponent` to a `<g>` inside another SVG it will append a `<svg>` inside that `<g>`?